benchmarks/density_ramp.py

```
@@ -0,0 +1,432 @@
+    try:
+        log_path = vm.data_dir / f"{vm.vm_id}.log"
+        if log_path.exists():
+            lines = log_path.read_text(errors="replace").splitlines()
+            return "\n".join(lines[-tail_lines:]) if lines else "(log file is empty)"
+        return f"(log not found: {log_path})"
+    except Exception as exc:
+        return f"(could not read log: {exc})"
+
+
+def boot_one(
+    seq: int,
```

Ruff (0.15.4)

[warning] Unused function argument: `seq`

(ARG001)